src/clients/SpokePoolClient.ts

```diff
@@ -677,16 +686,30 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @note This method is the primary method for updating the state of this client externally.
    * @see _update
    */
-  public async update(eventsToQuery = this.queryableEventNames): Promise<void> {
+  public async update(eventsToQuery = this.queryableEventNames, timeout = Number.POSITIVE_INFINITY): Promise<void> {
     if (this.hubPoolClient !== null && !this.hubPoolClient.isUpdated) {
       throw new Error("HubPoolClient not updated");
     }
 
-    const update = await this._update(eventsToQuery);
+    const updates = [this._update(eventsToQuery)];
+
+    // If a timeout has been specified and it occurs first, _update() will continue in the background. It
+    // will ultimately resolve but it does not touch SpokePoolClient state so its result will never be used.
+    if (timeout !== Number.POSITIVE_INFINITY) {
+      assert(timeout > 0);
+      const updateTimeout = async (timeout: number): Promise<SpokePoolUpdate> => {
+        await delay(timeout);
+        return { success: false, reason: `Timed out after ${timeout} seconds` };
+      };
+      updates.push(updateTimeout(timeout));
+    }
+
+    const update: SpokePoolUpdate = await Promise.race(updates);
```

Comment:
todo: Need to wrap this in a `try...catch` to also handle RPC errors.

Reply:
Fixed: https://github.com/across-protocol/sdk-v2/pull/611/commits/9282a96b8ab0ee8fb34ebcae21490fec7327ffe9.